bin/lib/zopen-build

```diff
@@ -343,33 +367,73 @@ checkEnv()
   #
   printHeader "Checking environment configuration"
 
-
-  if [ ! -z "$ZOPEN_FORCE_TYPE" ]; then
-    # user specified, so normalize in upper case
-    ZOPEN_TYPE=$(echo "${ZOPEN_FORCE_TYPE}" | tr '[a-z]' '[A-Z]')
-  fi
-
-  if [ "${ZOPEN_TYPE}x" = "TARBALLx" ]; then
-    if [ "${ZOPEN_TARBALL_URL}x" = "x" ]; then
-      export ZOPEN_TARBALL_URL="${ZOPEN_URL}"
-    fi
-    if [ "${ZOPEN_TARBALL_DEPS}x" = "x" ]; then
-      export ZOPEN_TARBALL_DEPS="${ZOPEN_DEPS}"
-    fi
-    printHeader "Using tarball environment"
-  elif [ "${ZOPEN_TYPE}x" = "GITx" ]; then
-    if [ "${ZOPEN_GIT_URL}x" = "x" ]; then
-      export ZOPEN_GIT_URL="${ZOPEN_URL}"
-    fi
-
-    if [ "${ZOPEN_GIT_DEPS}x" = "x" ]; then
-      export ZOPEN_GIT_DEPS="${ZOPEN_DEPS}"
+  if [ ! -z "$ZOPEN_BUILD_LEVEL" ]; then
+    if [ "${ZOPEN_BUILD_LEVEL}x" = "DEVx" ]; then
+      if [ -z "$ZOPEN_DEV_DEPS" ]; then
+        printError "Building from dev, but ZOPEN_DEV_DEPS not specified"
+      fi
+      ZOPEN_DEPS="$ZOPEN_DEV_DEPS"  
+      if [ -z "$ZOPEN_DEV_URL" ]; then
+        printError "Building from dev, but ZOPEN_DEV_URL not specified"
+      fi
+      ZOPEN_URL="$ZOPEN_DEV_URL"  
+      if [ ! -z "$ZOPEN_DEV_BRANCH" ]; then
+        export ZOPEN_GIT_BRANCH="$ZOPEN_DEV_BRANCH"
+      fi
+      if [ ! -z "$ZOPEN_DEV_TAG" ]; then
+        export ZOPEN_GIT_TAG="$ZOPEN_DEV_TAG"
+      fi
+    elif [ "${ZOPEN_BUILD_LEVEL}x" = "STABLEx" ]; then
+      if [ -z "$ZOPEN_STABLE_DEPS" ]; then
+        printError "Building from stable, but ZOPEN_STABLE_DEPS not specified"
+      fi
+      ZOPEN_DEPS="$ZOPEN_STABLE_DEPS"  
+      if [ -z "$ZOPEN_STABLE_URL" ]; then
+        printError "Building from stable, but ZOPEN_STABLE_URL not specified"
+      fi
+      ZOPEN_URL="$ZOPEN_STABLE_URL"  
+      if [ ! -z "$ZOPEN_STABLE_BRANCH" ]; then
+        export ZOPEN_GIT_BRANCH="$ZOPEN_STABLE_BRANCH"
+      fi
+      if [ ! -z "$ZOPEN_STABLE_TAG" ]; then
+        export ZOPEN_GIT_TAG="$ZOPEN_STABLE_TAG"
+      fi
+    else
+      printError "Unrecognized build level ${ZOPEN_BUILD_LEVEL}"
+    fi 
+    extension="${ZOPEN_URL##*.}"
+    if [ "$extension" = "git" ]; then
```

Comment:
nice.

Reply:
The doc should be updated to change ZOPEN_TYPE from being _required_ to being _ignored_. 
Do you want to add a check that validates that IF ZOPEN_GIT is already specified, that it's consistent with what you are about to set it to? And maybe spit out a WARNING indicating the variable should be removed?